fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/EliminateAliasNode.java

```diff
@@ -0,0 +1,91 @@
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.logical.LogicalPlan;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Eliminate the logical sub query and alias node after analyze and before rewrite
+ * If we match the alias node and return its child node, in the execute() of the job
+ * <p>
+ * TODO: refactor group merge strategy to support the feature above
+ */
+public class EliminateAliasNode implements AnalysisRuleFactory {
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+                RuleType.PROJECT_ELIMINATE_ALIAS_NODE.build(
+                        logicalProject(logicalSubQueryAlias())
+                                .then(project -> eliminateSubQueryAliasNode(project, project.children()))
+                ),
+                RuleType.FILTER_ELIMINATE_ALIAS_NODE.build(
+                        logicalFilter(logicalSubQueryAlias())
+                                .then(filter -> eliminateSubQueryAliasNode(filter, filter.children()))
+                ),
+                RuleType.AGGREGATE_ELIMINATE_ALIAS_NODE.build(
+                        logicalAggregate(logicalSubQueryAlias())
+                                .then(agg -> eliminateSubQueryAliasNode(agg, agg.children()))
+                ),
+                RuleType.JOIN_ELIMINATE_ALIAS_NODE.build(
+                        logicalJoin().then(join -> joinEliminateSubQueryAliasNode(join, join.children()))
+                )
+        );
+    }
+
+    private LogicalPlan eliminateSubQueryAliasNode(LogicalPlan node, List<Plan> aliasNodes) {
+        List<Plan> nodes = aliasNodes.stream()
+                .map(this::getPlan)
+                .collect(Collectors.toList());
+        return (LogicalPlan) node.withChildren(nodes);
+    }
+
+    private LogicalPlan joinEliminateSubQueryAliasNode(LogicalPlan node, List<Plan> aliasNode) {
+        List<Plan> nodes = aliasNode.stream()
+                .map(child -> {
+                    if (checkIsSubQueryAliasNode((GroupPlan) child)) {
+                        return ((GroupPlan) child).getGroup()
+                                .getLogicalExpression()
+                                .child(0)
+                                .getLogicalExpression()
+                                .getPlan();
+                    }
+                    return child;
+                })
+                .collect(Collectors.toList());
+        return (LogicalPlan) node.withChildren(nodes);
+    }
+
+    private boolean checkIsSubQueryAliasNode(GroupPlan node) {
+        return node.getGroup().getLogicalExpression().getPlan().getType()
+                == PlanType.LOGICAL_SUBQUERY_ALIAS;
+    }
+
+    private Plan getPlan(Plan node) {
+        return ((GroupPlan) node.child(0)).getGroup().getLogicalExpression().getPlan();
```

Comment:
get children plan from GroupPlan is not recommended.

I think you can simple return LogicalSubQueryAlias's children plan which is groupPlan.

```java
logicalProject(logicalSubQueryAlias()).then(project -> {
  return project.withChildren(project.child().child());
});

logicalJoin(logicalSubQueryAlias(), groupPlan()).then(join -> {
  return join.withChildren(ImmutableList.of(join.left().child(), join.right()));
});

logicalJoin(groupPlan(), logicalSubQueryAlias()).then(join -> {
  return join.withChildren(ImmutableList.of(join.left(), join.right().child()));
});
```

@morrySnow do you think so?

Reply:
I've tried this, but unluckily it will core at Select * From T1 T;

Reply:
It works, I've rewritten the code, thx.